modules/adomikAnalyticsAdapter.js

```diff
@@ -0,0 +1,202 @@
+  },
+  formatSize: (typedEventSize) => {
+    return {
+      width: Number(typedEventSize.width),
+      height: Number(typedEventSize.height)
+    };
+  },
+  handleSize: (sizes, typedEventSize) => {
+    let formattedSize = null;
+    if (sizeUtils.sizeAlreadyExists(sizes, typedEventSize) === undefined) {
+      formattedSize = sizeUtils.formatSize(typedEventSize);
+    }
+    return formattedSize;
+  }
+};
+
+const bucketEvents = [];
+
+function buildTypedEvents() {
+  const groupedTypedEvents = [];
+  bucketEvents.forEach(function(typedEvent, i) {
+    const [placementCode, type] = [typedEvent.event.placementCode, typedEvent.type];
+    let existTypedEvent = groupedTypedEvents.findIndex((groupedTypedEvent) => groupedTypedEvent.placementCode === placementCode);
+
+    if (existTypedEvent === -1) {
+      groupedTypedEvents.push({
+        placementCode: placementCode,
+        [type]: [typedEvent]
+      });
+      existTypedEvent = groupedTypedEvents.length - 1;
+    }
+
+    if (groupedTypedEvents[existTypedEvent][type]) {
+      groupedTypedEvents[existTypedEvent][type] = [...groupedTypedEvents[existTypedEvent][type], typedEvent];
+    } else {
+      groupedTypedEvents[existTypedEvent][type] = [typedEvent];
+    }
+  });
+
+  return groupedTypedEvents;
+}
+
+function sendTypedEvent() {
+  const groupedTypedEvents = buildTypedEvents();
+
+  const bulkEvents = {
+    uid: currentContext.uid,
+    ahbaid: currentContext.id,
+    timeout: currentContext.timeout,
+    hostname: window.location.hostname,
+    eventsByPlacementCode: groupedTypedEvents.map(function(typedEventsByType) {
+      let sizes = [];
+      const eventKeys = ['request', 'response', 'winner'];
+      let events = {};
+
+      eventKeys.forEach((eventKey) => {
+        events[`${eventKey}s`] = [];
+        if (typedEventsByType[eventKey] !== undefined) {
+          typedEventsByType[eventKey].forEach((typedEvent) => {
+            if (typedEvent.event.size !== undefined) {
+              const size = sizeUtils.handleSize(sizes, typedEvent.event.size);
+              if (size !== null) {
+                sizes = [...sizes, size];
+              }
+            }
+            events[`${eventKey}s`] = [...events[`${eventKey}s`], typedEvent.event];
+          });
+        }
+      });
+
+      return {
+        placementCode: typedEventsByType.placementCode,
+        sizes,
+        events
+      };
+    })
+  };
+
+  // Encode object in base64
+  const encodedBuf = window.btoa(JSON.stringify(bulkEvents));
+  // Create final url and split it in 1600 characters max (+endpoint length)
+  const encodedUri = encodeURIComponent(encodedBuf);
+
+  const splittedUrl = encodedUri.match(/.{1,1600}/g);
+  splittedUrl.forEach((split, i) => {
+    const partUrl = `${split}&id=${currentContext.id}&part=${i}&on=${splittedUrl.length - 1}`;
+    const img = new Image(1, 1);
+    img.src = 'http://' + currentContext.url + '/?q=' + partUrl;
```

Comment:
Are you sure that this makes a request on all the browsers you care about?

I'm cool with it if you are... but most people would use [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest). I'm not sure if anything in the HTML spec guarantees that the browser makes the request just by setting the src on an image, so it might throw off your stats.